File: python/chapter9/ice_cream_stand.py
class Restaurant:
    def __init__(self, name: str, cuisine_type: str, number_served: int):
        self.name = name
        self.cuisine_type = cuisine_type
        self.number_served = number_served

    def describeRestaurant(self):
        print(f"Name: {self.name}")
        print(f"Cuisine Type: {self.cuisine_type}")

    def openRestaurant(self):
        print(f"The {self.name} Restaurant is now open!")

    def setNumberServed(self, num: int):
        self.number_served = num

    def incrementNumberServed(self, num: int):
        self.number_served += num


class IceCreamStand(Restaurant):
    def __init__(self, flavors: list[str]):
        self.flavors = flavors

    def show_flavors(self):
        for flavor in self.flavors:
            print(flavor)


ics = IceCreamStand(["vanilla", "chocolate", "cream"])
ics.show_flavors()

File: javascript/chapter9/ice_cream_stand.js
class Restaurant {
	/**
	 * @param {string} name
	 * @param {string} cuisineType
	 */
	constructor(name, cuisineType) {
		this.name = name;
		this.cuisineType = cuisineType;
		this.numberServed = 0;
	}

	describeRestaurant() {
		console.log("Name: %s\n", this.name);
		console.log("Cuisine Type: %s\n", this.cuisineType);
	}

	openRestaurant() {
		console.log("The %s Restaurant is now open!\n", this.name);
	}

	/** @param {number} num */
	setNumberServed(num) {
		this.numberServed = num;
	}

	/** @param {number} num */
	incrementNumberServed(num) {
		this.numberServed += num;
	}
}

class IceCreamStand extends Restaurant {
	/** @param {string[]} flavors */
	constructor(flavors) {
		super("", "");
		this.flavors = flavors;
	}

	showFlavors() {
		for (const flavor of this.flavors) console.log(flavor);
	}
}

const ics = new IceCreamStand(["vanilla", "chocolate", "cream"]);
ics.showFlavors();
